src/internal.c

```diff
@@ -7386,24 +7385,29 @@ int SendUserAuthRequest(WOLFSSH* ssh, byte authId, int addSig)
             WLOG(WS_LOG_DEBUG, "SUAR: Calling the userauth callback");
 
             WMEMSET(&authData, 0, sizeof(authData));
-            authData.type = authId;
 
-            if (authId == ID_USERAUTH_PASSWORD) {
+            authData.username = (const byte*)ssh->userName;
+            authData.usernameSz = ssh->userNameSz;
+
+            ret = ssh->ctx->userAuthCb(WOLFSSH_USERAUTH_PUBLICKEY,
```

Comment:
automatically tries calling user callback with public key type first, if callback rejects it then tries with password